### ISSUE TITLE
Cannot view dashboards as public user

### ISSUE THREAD

**Issue by sholtoarmstrong-iot** [apache/superset#16718]
I set the public user permissions to allow all the permissions I could find. I then got the public link and added "?standalone=true". I am not able to view the dashboard.

### Expected results

I expect to be able to see the dashboards as a public user with the correct permissions.

### Actual results

I get a very non-descriptive error or the frontend.

#### Screenshots

![Screenshot from 2021-09-16 15-17-53](https://user-images.githubusercontent.com/65215432/133619233-8ba1985c-e238-4fa3-a6ee-a0f206d0fbd0.png)

#### How to reproduce the bug

1. Use the pre-configured demo charts
2. Add the permissions to the public user
3. Get the public link of the dashboard "https://xxxx/r/1?standalone=true"
4. Logout and visit link. Presented with an error.

### Environment

(please complete the following information):

Using the docker image: apache/superset:1.3.1rc1 with the helm deploy
- superset version: `1.3.1-rc1`
- python version: `Python 3.7.9`
- node.js version: Cant find node but its the version installed on the image
- any feature flags active:
The following config overrides were used with the deploy to allow it to run in an iframe and enable some features.
```
configOverrides:

  allow_iframe_access: |

    HTTP_HEADERS = {'X-Frame-Options': 'ALLOWALL', 'Access-Control-Allow-Origin': '*', 'X-Forwarded-Proto': 'https'}
    ENABLE_PROXY_FIX = True
    WTF_CSRF_ENABLED = False
  feature_flags: |
    FEATURE_FLAGS = {
      "DASHBOARD_CROSS_FILTERS": True,
      "DASHBOARD_NATIVE_FILTERS": True,
      "ALERT_REPORTS": True
    }
```

### Additional context
This used to work perfectly fine with apache/superset:1.1.0 and now does not seem to.
Logs from the server
```
[2021-09-16 13:09:14 +0000] [10] [INFO] Starting gunicorn 20.0.4
[2021-09-16 13:09:14 +0000] [10] [INFO] Listening at: http://0.0.0.0:8088 (10)
[2021-09-16 13:09:14 +0000] [10] [INFO] Using worker: gthread
[2021-09-16 13:09:14 +0000] [13] [INFO] Booting worker with pid: 13
logging was configured successfully
2021-09-16 13:09:16,831:INFO:superset.utils.logging_configurator:logging was configured successfully
2021-09-16 13:09:16,838:INFO:root:Configured event logger of type <class 'superset.utils.log.DBEventLogger'>
2021-09-16 13:14:02,573:WARNING:root:Class 'werkzeug.local.LocalProxy' is not mapped
Loaded your LOCAL configuration at [/app/pythonpath/superset_config.py]
172.31.14.195 - - [16/Sep/2021:13:14:02 +0000] "GET /superset/dashboard/10/?preselect_filters=%7B%7D&native_filters=%28%27104%27%3A%28extraFormData%3A%28%29%2CfilterState%3A%28value%3A%21n%29%2Cid%3A%27104%27%2CownState%3A%28%29%29%29 HTTP/1.1" 200 26709 "-" "Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/91.0.4472.77 Safari/537.36"
172.31.14.195 - - [16/Sep/2021:13:14:04 +0000] "GET /static/assets/images/favicon.png HTTP/1.1" 200 0 "https://xxxxxxx/superset/dashboard/10/?preselect_filters=%7B%7D&native_filters=%28%27104%27%3A%28extraFormData%3A%28%29%2CfilterState%3A%28value%3A%21n%29%2Cid%3A%27104%27%2CownState%3A%28%29%29%29" "Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/91.0.4472.77 Safari/537.36"
2021-09-16 13:14:04,216:WARNING:root:Class 'werkzeug.local.LocalProxy' is not mapped
172.31.14.195 - - [16/Sep/2021:13:14:04 +0000] "GET /api/v1/dashboard/10/charts HTTP/1.1" 200 1483 "https://xxxxxxx/superset/dashboard/10/?preselect_filters=%7B%7D&native_filters=%28%27104%27%3A%28extraFormData%3A%28%29%2CfilterState%3A%28value%3A%21n%29%2Cid%3A%27104%27%2CownState%3A%28%29%29%29" "Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/91.0.4472.77 Safari/537.36"
2021-09-16 13:14:04,247:WARNING:root:Class 'werkzeug.local.LocalProxy' is not mapped
172.31.14.195 - - [16/Sep/2021:13:14:04 +0000] "GET /api/v1/dashboard/10 HTTP/1.1" 200 2473 "https://xxxxxxx/superset/dashboard/10/?preselect_filters=%7B%7D&native_filters=%28%27104%27%3A%28extraFormData%3A%28%29%2CfilterState%3A%28value%3A%21n%29%2Cid%3A%27104%27%2CownState%3A%28%29%29%29" "Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/91.0.4472.77 Safari/537.36"
2021-09-16 13:14:04,339:WARNING:root:Class 'werkzeug.local.LocalProxy' is not mapped
172.31.14.195 - - [16/Sep/2021:13:14:04 +0000] "GET /api/v1/dashboard/10/datasets HTTP/1.1" 200 1316 "https://xxxxxxx/superset/dashboard/10/?preselect_filters=%7B%7D&native_filters=%28%27104%27%3A%28extraFormData%3A%28%29%2CfilterState%3A%28value%3A%21n%29%2Cid%3A%27104%27%2CownState%3A%28%29%29%29" "Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/91.0.4472.77 Safari/537.36"
172.31.14.195 - - [16/Sep/2021:13:14:05 +0000] "GET /csstemplateasyncmodelview/api/read HTTP/1.1" 200 625 "https://xxxxxxx/superset/dashboard/10/?preselect_filters=%7B%7D&native_filters=%28%27104%27%3A%28extraFormData%3A%28%29%2CfilterState%3A%28value%3A%21n%29%2Cid%3A%27104%27%2CownState%3A%28%29%29%29" "Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/91.0.4472.77 Safari/537.36"
2021-09-16 13:14:06,205:WARNING:root:Class 'werkzeug.local.LocalProxy' is not mapped
172.31.14.195 - - [16/Sep/2021:13:14:06 +0000] "POST /superset/log/?explode=events&dashboard_id=10 HTTP/1.1" 200 1 "https://xxxxxxx/superset/dashboard/10/?preselect_filters=%7B%7D&native_filters=%28%27104%27%3A%28extraFormData%3A%28%29%2CfilterState%3A%28value%3A%21n%29%2Cid%3A%27104%27%2CownState%3A%28%29%29%29" "Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/91.0.4472.77 Safari/537.36"
```
Logs from the worker
`
logging was configured successfully
2021-09-16 13:09:16,979:INFO:superset.utils.logging_configurator:logging was configured successfully
2021-09-16 13:09:16,988:INFO:root:Configured event logger of type <class 'superset.utils.log.DBEventLogger'>

`

**Comment by panqingdian** [comment]
Mine has the same problem and the same error. What can I do to solve this problem? I don't know. Did you handle it

**Comment by isavans** [comment]
@mayurnewase I have been trying to find a simple solution for this issue.
I cant get this to work if disable flag ENABLE_REACT_CRUD_VIEWS and ALERT_REPORTS.
But that is not an optimal solution since then the dashboard list looks like this
![image](https://user-images.githubusercontent.com/92523108/145021428-32c6a3e4-c191-4555-bec4-95c96e4ae3ea.png)

Instead of this
![image](https://user-images.githubusercontent.com/92523108/145021711-e03568de-77f9-4aef-9838-730b0b0579d5.png)

I can see that your patch (https://github.com/apache/superset/pull/17106) has been merged into master.
Do you know if it is in release 1.4.0?

**Comment by isavans** [comment]
This is still an issue in version 1.4.0.
Does anyone have an idea what version will include the fix for this.

**Comment by sfirke** [comment]
The public, i.e. people not logged in, can view dashboards in version 2.0.0.  I have `DASHBOARD_RBAC` feature flag enabled and the `Public` role assigned under dashboard properties -> Roles.  Because it's mentioned above, I also have `ALERT_REPORTS` flag enabled too.

IMO this can be closed as fixed in 2.0.0.

**Comment by MM-Lehmann** [comment]
This always worked imho, but can others also access the dashboard purely based on the data access permissions?
  
Especially on a different dashboard, that hasn't assigned any roles access?
  
That's the part that doesn't work for me.
  

  

  
  
  
  
>

**Comment by sfirke** [comment]
> Can others also access the dashboard purely based on the data access permissions? Especially on a different dashboard, that hasn't assigned any roles access?

Not sure, my use case doesn't do much with data access permissions and I've been assigning the Public role to all public-facing dashboards.

**Comment by eschutho** [comment]
> IMO this can be closed as fixed in 2.0.0.

Closing this as per the comment above.
